src/Services/ArrayInterface.php

```diff
@@ -6,32 +6,194 @@
+     * @return array
+     */
+    public function only(array $array, ...$requiredKeys): array;
+
+    /**
+     * method returns bool whether any item in given set exists in an array using "dot" notation
+     *
+     * @param array $arr
+     * @param mixed $keys
+     * @param bool $strictCheck
+     * @return bool
+     */
+    public function has(array $arr, $keys, $strictCheck = true): bool;
+
+    /**
+     * method returns bool when the given key exists in an array
+     *
      * @param array $arr
      * @param string $key
      * @return bool
      */
     public function hasKey(array $arr, string $key): bool;
 
+    /**
+     * method removes the given key/value pair from an array
+     *
+     * @param array $arr
+     * @param mixed ...$keys
+     * @return array
+     */
+    public function remove(array $arr, ...$keys): array;
+
+    /**
+     * method returns the match elements of an array using a callback function
+     *
+     * @param array $arr
+     * @param callable $callback
+     * @return array
+     */
+    public function match(array $arr, callable $callback): array;
+
+
+    /**
+     * method sorts an array by its values and it also sort by the given closure
+     *
+     * @param array $arr
+     * @param callable|null $callable
+     * @return array
+     */
+    public function sort(array $arr, callable $callable = null): array; // TODO: not needed.
+
+    /**
+     * method randomly shuffles the items in the array
+     *
+     * @param array $arr
+     * @return array
+     */
+    public function shuffle(array $arr): array;   //TODO: not needed.
+
+    /**
+     * method Removes the element  from an array at the specified position
+     *
+     * @param array $arr
+     * @param int $position
+     * @return void
+     */
+    public function pop(array &$arr, int $position): void;
+
+    /**
+     * method returns and removes a key value pair from an array
+     * default value may be passed as the third argument to the method
+     * This value will be returned if the key doesn't exist
+     *
+     * @param array $arr
+     * @param int|string $key
+     * @param int|string $default
+     * @return mixed
+     */
+    public function pull(array $arr, $key, $default);
+
+    /**
+     * method returns a random value/values from an array
+     *
+     * @param array $arr
+     * @param int $itemsCount
+     * @return mixed
+     */
+    public function random(array $arr, int $itemsCount);
+
+    /**
+     * method retrieves all of the values for a given key from an array using "dot" notation
+     *
+     * @param array $arr
+     * @param mixed ...$keys
+     * @return array
+     */
+    public function pickN(array $arr, ...$keys): array;
+
+    /**
+     * method removes a given key/value pair from a nested array using "dot" notation
+     *
+     * @param array $arr
+     * @param string $str
+     * @return array
+     */
+    public function forgetN(array $arr, string $str): array;
```

Comment:
I wish we could create better name for this method.

Reply:
remove this method